zebra-consensus/src/primitives/groth16/params.rs

```diff
@@ -35,49 +63,42 @@ pub struct SaplingParams {
 }
 
 impl SaplingParams {
-    fn new() -> Self {
-        let (spend, output) = wagyu_zcash_parameters::load_sapling_parameters();
-        let spend_fs = BufReader::with_capacity(1024 * 1024, &spend[..]);
-        let output_fs = BufReader::with_capacity(1024 * 1024, &output[..]);
-
-        Self::read(spend_fs, output_fs)
-            .expect("reading parameters from wagyu zcash parameter's vec will always succeed")
-    }
-
-    fn read<R: io::Read>(spend_fs: R, output_fs: R) -> Result<Self, io::Error> {
-        let mut spend_fs = HashReader::new(spend_fs);
-        let mut output_fs = HashReader::new(output_fs);
-
-        // Deserialize params
-        let spend = groth16::Parameters::<Bls12>::read(&mut spend_fs, false)?;
-        let output = groth16::Parameters::<Bls12>::read(&mut output_fs, false)?;
-
-        // There is extra stuff (the transcript) at the end of the parameter file which is
-        // used to verify the parameter validity, but we're not interested in that. We do
-        // want to read it, though, so that the BLAKE2b computed afterward is consistent
-        // with `b2sum` on the files.
-        let mut sink = io::sink();
-        io::copy(&mut spend_fs, &mut sink)?;
-        io::copy(&mut output_fs, &mut sink)?;
-
-        assert!(
-            spend_fs.into_hash() == SAPLING_SPEND_HASH,
-            "Sapling spend parameter is not correct."
-        );
-        assert!(
-            output_fs.into_hash() == SAPLING_OUTPUT_HASH,
-            "Sapling output parameter is not correct."
-        );
+    /// Download if needed, cache, check, and load the Sapling Groth16 parameters.
+    ///
+    /// # Panics
+    ///
+    /// If the downloaded or pre-existing parameter files are invalid.
+    fn new() -> SaplingParams {
+        // TODO: Sprout
```

Comment:
Minor: Should this `TODO` be outside this type? I.e., will we have a `SproutParams` type?

Reply:
I haven't made that decision yet, but whatever I do, the TODO will be deleted in the next PR.

Reply:
(The loader handles Sapling and Sprout in the same function call. So I might have to move the download code into the `Groth16Parameters` type.)